src/test/mir-opt/nrvo-simple/rustc.nrvo.RenameReturnPlace.diff

```diff
@@ -0,0 +1,41 @@
+- // MIR for `nrvo` before RenameReturnPlace
++ // MIR for `nrvo` after RenameReturnPlace
+  
+  fn nrvo(_1: for<'r> fn(&'r mut [u8; 1024])) -> [u8; 1024] {
+      debug init => _1;                    // in scope 0 at $DIR/nrvo-simple.rs:2:9: 2:13
+-     let mut _0: [u8; 1024];              // return place in scope 0 at $DIR/nrvo-simple.rs:2:39: 2:49
++     let mut _0: [u8; 1024];              // return place in scope 0 at $DIR/nrvo-simple.rs:3:9: 3:16
+      let mut _2: [u8; 1024];              // in scope 0 at $DIR/nrvo-simple.rs:3:9: 3:16
+      let _3: ();                          // in scope 0 at $DIR/nrvo-simple.rs:4:5: 4:19
+      let mut _4: for<'r> fn(&'r mut [u8; 1024]); // in scope 0 at $DIR/nrvo-simple.rs:4:5: 4:9
+      let mut _5: &mut [u8; 1024];         // in scope 0 at $DIR/nrvo-simple.rs:4:10: 4:18
+      let mut _6: &mut [u8; 1024];         // in scope 0 at $DIR/nrvo-simple.rs:4:10: 4:18
+      scope 1 {
+-         debug buf => _2;                 // in scope 1 at $DIR/nrvo-simple.rs:3:9: 3:16
++         debug buf => _0;                 // in scope 1 at $DIR/nrvo-simple.rs:3:9: 3:16
+      }
+  
+      bb0: {
+-         StorageLive(_2);                 // scope 0 at $DIR/nrvo-simple.rs:3:9: 3:16
+-         _2 = [const 0u8; 1024];          // scope 0 at $DIR/nrvo-simple.rs:3:19: 3:28
++         _0 = [const 0u8; 1024];          // scope 0 at $DIR/nrvo-simple.rs:3:19: 3:28
+                                           // ty::Const
+                                           // + ty: u8
+                                           // + val: Value(Scalar(0x00))
+                                           // mir::Constant
+                                           // + span: $DIR/nrvo-simple.rs:3:20: 3:21
+                                           // + literal: Const { ty: u8, val: Value(Scalar(0x00)) }
+          StorageLive(_3);                 // scope 1 at $DIR/nrvo-simple.rs:4:5: 4:19
+-         _6 = &mut _2;                    // scope 1 at $DIR/nrvo-simple.rs:4:10: 4:18
++         _6 = &mut _0;                    // scope 1 at $DIR/nrvo-simple.rs:4:10: 4:18
```

Comment:
can you run this test with miri? I think it should work, since the `return` terminator in miri copies the value in `_0` to whatever the call site wants to assign to. It's not the same memory, but I don't see how this change could break it. Still, better safe than sorry.

Reply:
Yes. A new `miri_unleahed` test? Are we supposed to write `run-pass` ones.

Reply:
I did add a new test with `// run-pass`, but this seems [not to be possible](https://github.com/rust-lang/rust/pull/72205/files#diff-91d471b6a33efa383acdf4afdf794766R26). I suppose I can write a simpler one that has the optimization applied with `#[inline(never)]` on a `const fn` to keep the test relevant once MIR inlining is enabled.

Reply:
Ah no, I meant in miri the tool, so in https://github.com/rust-lang/miri/tree/master/tests/run-pass and run `./x.py test src/tools/miri --stage 1`. There's no need to merge it in this PR, just open a PR against miri once you've confirmed it works.

Reply:
Correction: `./x.py test src/tools/miri --stage 0`.